evalml/pipelines/utils.py

```diff
@@ -364,14 +389,21 @@ def _make_pipeline_time_series(
         )
         pipeline = _make_pipeline_from_multiple_graphs(
             [pipeline, kina_pipeline],
-            estimator,
+            DropNaNRowsTransformer if need_drop_nan else estimator,
             problem_type,
             parameters=parameters,
             sub_pipeline_names={
                 kina_pipeline.name: "Known In Advance",
                 pipeline.name: "Not Known In Advance",
             },
         )
+        if need_drop_nan:
+            last_component_name = pipeline.component_graph.get_last_component().name
+            pipeline.component_graph.component_dict[estimator.name] = [
+                estimator,
+                last_component_name + ".x",
+                last_component_name + ".y",
+            ]
```

Comment:
I feel like this and lines 381-383 + 392 are a bit hacky, but I couldn't find a better function to do this.

Reply:
I think this is another example of why we need this: https://github.com/alteryx/evalml/issues/2997